src/test/java/it/pagopa/transactions/utils/UrlUtils.java

```diff
@@ -0,0 +1,55 @@
+package it.pagopa.transactions.utils;
+
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class UrlUtils {
+
+    public static boolean urlsEqualsWithRandomParam(
+                                                    String actual,
+                                                    String expected
+    ) {
+        URI actualUri = URI.create(actual);
+        URI expectedUri = URI.create(expected);
+
+        // Compare full path
+        if (!Objects.equals(actualUri.getHost(), expectedUri.getHost()) ||
+                actualUri.getPort() != expectedUri.getPort() ||
+                !actualUri.getPath().equals(expectedUri.getPath()))
+            return false;
+
+        // Compare fragments
+        if (!getParametersAsMap(actualUri.getFragment()).equals(getParametersAsMap(expectedUri.getFragment())))
+            return false;
+
+        // Compare query params without random, if are equals complete url is equals
+        return getParametersAsMap(actualUri.getQuery(), true).equals(getParametersAsMap(expectedUri.getQuery(), true));
+    }
+
+    public static Map<String, String> getParametersAsMap(String query) {
+        return getParametersAsMap(query, false);
+    }
+
+    public static Map<String, String> getParametersAsMap(
+                                                         String query,
+                                                         boolean removeRandom
+    ) {
+        if (query == null || query.isEmpty()) {
```

Comment:
Instead of checking for `null `directly, you can use `Optional `to handle potential `null `values more gracefully.

   ```
 public static Map<String, String> getParametersAsMap(String query, boolean removeRandom) {
        return Optional.ofNullable(query)
                       .map(q -> Arrays.stream(q.split("&"))
                                       .map(param -> param.split("="))
                                       .filter(pair -> !removeRandom || !pair[0].equals("t"))
                                       .collect(Collectors.toMap(
                                           pair -> pair[0],
                                           pair -> pair.length > 1 ? pair[1] : ""
                                       )))
                       .orElse(Map.of());
    }
```

Reply:
Done here ffce41c35a3c04e52f19b7bd8478fcea89aa893e.
But in this suggestion is missing empty string check, without this a map with a wrong value would be returned.

Reply:
you a right, try something like this?


`public static Map<String, String> getParametersAsMap(String query, boolean removeRandom) {
    return Optional.ofNullable(query)
                   .filter(q -> !q.isEmpty()) // Check if the query is not empty
                   .map(q -> Arrays.stream(q.split("&"))
                                   .map(param -> param.split("="))
                                   .filter(pair -> !removeRandom || !pair[0].equals("t"))
                                   .collect(Collectors.toMap(
                                       pair -> pair[0],
                                       pair -> pair.length > 1 ? pair[1] : ""
                                   )))
                   .orElse(Map.of());
}
`

Reply:
Yes is already done in the previous commit